geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/AbstractRedisData.java

```diff
@@ -241,6 +243,11 @@ public void fromDelta(DataInput in) throws IOException, InvalidDeltaException {
       case SET_BYTE_ARRAY_AND_TIMESTAMP:
         SetByteArrayAndTimestamp.deserializeFrom(in, this);
         break;
+      case REPLACE_BYTE_ARRAYS:
+        synchronized (this) {
```

Comment:
Given the other changes you are making (i.e. replacing the MemberSet with a brand new one instead of clearing and adding the byte arrays to an existing MemberSet) you no longer need to add this "synchronized".